crates/transaction-pool/src/lib.rs

```diff
@@ -473,6 +473,18 @@ where
         self.pool.get_transactions_by_sender(sender)
     }
 
+    fn get_transactions_by_sender_and_nonce(
+        &self,
+        sender: Address,
+        nonce: u64,
+    ) -> Option<Arc<ValidPoolTransaction<Self::Transaction>>> {
+        let transaction_id = TransactionId::new(self.pool.get_sender_id(sender), nonce);
+
+        let AllPoolTransactions { pending, queued } = self.all_transactions();
+
+        pending.iter().chain(queued.iter()).find(|tx| tx.transaction_id == transaction_id).cloned()
+    }
```

Comment:
sorry, I meant this field here:

https://github.com/paradigmxyz/reth/blob/8dff31e3accf63521f746dc797ce0c3da2ea86bf/crates/transaction-pool/src/pool/txpool.rs#L886-L887

inside the TxPool struct

https://github.com/paradigmxyz/reth/blob/8dff31e3accf63521f746dc797ce0c3da2ea86bf/crates/transaction-pool/src/pool/txpool.rs#L70-L71

see also this function we currently only use in tests

https://github.com/paradigmxyz/reth/blob/8dff31e3accf63521f746dc797ce0c3da2ea86bf/crates/transaction-pool/src/pool/txpool.rs#L931-L935

Reply:
I've thought about that but in the `PoolInner` structure, the `pool` field is private. So we can use the `pooled_transactions` method: 

https://github.com/paradigmxyz/reth/blob/8dff31e3accf63521f746dc797ce0c3da2ea86bf/crates/transaction-pool/src/pool/mod.rs#L289-L292

But in this case it also creates a vector of all transactions and so the benefit you described is lost. As a result, what I did was:
1. I added a method `get_pool_data` in `PoolInner` which reads the pool directly. I use it to read the pool here but given the number of times `self.pool.read()` is used in the `PoolInner` implementation, I think there is some refactoring to be done in another PR (I can take care of it next) using the newly created method directly.
2. I removed the cfg tests for the `get` method in the `TxPool` as it is now used to get from the transaction ID.

Tell me what you think and sorry for the back and forth :)

Reply:
> Tell me what you think and sorry for the back and forth :)

all good, appreciate the patience :)

I like the rwlock guard function.